src/BUILD

```diff
@@ -0,0 +1,83 @@
+load("@rules_rust//rust:defs.bzl", "rust_clippy", "rust_binary", "rustfmt_test")
+load("@rules_rust//crate_universe:defs.bzl", "crate", "crates_repository")
+
+OBJCOPY = "riscv64-unknown-elf-objcopy -I binary -O elf64-littleriscv"
+
+config_setting(
+    name = "debug_mode",
+    values = {
+    "compilation_mode": "dbg",
+    }
+)
+
+config_setting(
+    name = "optimized_mode",
+    values = {
+    "compilation_mode": "opt",
+    }
+)
+
+genrule(
+    name = "umode_to_object",
+    srcs = [
+        "//u-mode:umode"
+    ],
+    outs = ["umode.o"],
+    cmd = OBJCOPY + " $< $@",
```

Comment:
This is a dependency on the host CC?

Reply:
Yes. Is there another way to get this?

Reply:
I don't thing the method Cargo-binutils uses will work with bazel, at least not cleanly. Looking into how to use/extend the toolchain to do this more correctly.

Reply:
Fixed.